tests/__tests__/rely/StringMatchers_test.re

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */;
+open TestFramework;
+open MatcherSnapshotTestRunner;
+
+/* String matchers are also integration tested in depth in TestRunner_test.re*/
+describe("String matchers", ({test}) =>
+  test(
+    "expect.string.toMatch matches shouldn't have to be at the beginning ",
+    ({expect}) => {
+    let myString = "TEST2, TEST, TEST3";
+
+    expect.string(myString).toMatch("TEST");
```

Comment:
I'm not sure if this is desirable behavior. It's not clear whether the string needs to match the entire regex or the string just needs to contain the regex.

Would names like this be better?

- `expect.string(myString).toMatchRegex(".*Test.*"); // true`
- `expect.string(myString).toMatchRegex("Test"); // false`
- `expect.string(myString).toContainRegex("Test"); // true`
- `expect.string(myString).toContainString("Test"); // true`

Reply:
Or perhaps leave `toMatch` as you have it and add a `toMatchAll` or something.

Reply:
I'm also not sure, but this matches jest now, so I think it's probably right just to avoid bikeshedding. I think toContain is largely redundant with the new behavior, I could definitely see adding/renaming things, but I think that's outside the scope of this PR?